app/src/main/java/com/wut/indoornavigation/di/module/ApplicationModule.java

```diff
@@ -28,4 +29,10 @@ public ApplicationModule(Context context) {
     MapEngine provideMapEngine(CanvasExtender canvasExtender) {
         return new MapEngineImpl(context, canvasExtender);
     }
+
+    @Singleton
+    @Provides
+    StrategyProvider provideStrategyProvider() {
```

Comment:
mozesz to usunac. Ogolnie jak zrobilesz @Inject nad kontruktorem klasy to taki provider zostanie wygenerowany automatycznie. Providery trzeba glownie po to zeby wstrzykiwac interfejsy